### PR TITLE
Fix: make report independent of samples in LIMS

### PR DESCRIPTION
**Pull request by ivadym**
## Description

The delivery report generation fails for downsampled analysis since its not being able to extract the Apptags when there is no associated samples in LIMS.

> Could not fetch sample surelyfreshasp from LIMS: 404: Sample not found: surelyfreshasp
> ...
>   File "/home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages/cg/meta/report/report_api.py", line 311, in get_sample_application_data
>     tag=application.tag,
> AttributeError: 'NoneType' object has no attribute 'tag'

With this fix it will return an empty application object for this cases.

### Fixed
- Delivery report for downsamples analysis.

### Changed
- Delivery report sample name and sex retrieved from StatusDB and not LIMS.

### How to prepare for test
- [ ] ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] paxa the environment: `paxa`
- [ ] install on stage (example for hasta):
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh [THIS-BRANCH-NAME]`

---

### Tests

No force flag g for an analysis with downsamples samples:
Screen Shot 2022-10-05 at 15 01 37

Report generated with the force flag:
Screen Shot 2022-10-05 at 14 33 22

Screen Shot 2022-10-05 at 14 32 36

<details>
<summary>MIP-DNA (original report)</summary>
Screen Shot 2022-10-05 at 14 38 36
</details>

<details>
<summary>MIP-DNA (new report)</summary>
Screen Shot 2022-10-05 at 14 38 45
</details>
---

## Review
- [ ] code approved by
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by ivadym** [comment]
Could someone from @Clinical-Genomics/infrastructure have a quick look into this. Thanks :)

**Review by karlnyr — APPROVED** [review]
Except the comment I made I think this looks good! There is of course the discussion to be held elsewhere if we should always generate a delivery report for downsampled cases or if that should be a special function.

File: cg/meta/report/report_api.py [hunk 1]
```diff
@@ -273,11 +273,11 @@ def get_samples_data(
```
[the remark this answers is not in the document]
Comment (ivadym) [reply cg/meta/report/report_api.py:273]: 👍

**Comment by ivadym** [comment]
Thanks for the review @karlnyr!

> Except the comment I made I think this looks good! There is of course the discussion to be held elsewhere if we should always generate a delivery report for downsampled cases or if that should be a special function.

Tagging you @moahaegglund, maybe we can include this in the discussion of future updates of the delivery report.

**Comment by moahaegglund** [comment]
> Thanks for the review @karlnyr!
> 
> > Except the comment I made I think this looks good! There is of course the discussion to be held elsewhere if we should always generate a delivery report for downsampled cases or if that should be a special function.
> 
> Tagging you @moahaegglund, maybe we can include this in the discussion of future updates of the delivery report.

Perfect, thank you! Yeah, let's think about if we should create a delivery report or not in these cases.

**Pull request merged**